pasta/augment/import_utils.py

```diff
@@ -53,3 +55,34 @@ def split_import(sc, node, alias_to_remove):
 
   parent_list.insert(idx + 1, new_import)
   return new_import
+
+def get_unused_imports(tree):
+  """Get the import nodes that aren't used.
+
+  Arguments:
+    tree: (ast.AST) An ast tree to find imports in.
+
+  Returns:
+    A list of ast.AST representing nodes that are imports or parts of imports
+    (aliases) that can be removed because they are unused.
+  """
+  sc = scope.analyze(tree)
+  unused_aliases = set()
+  for node in ast.walk(tree):
+    if isinstance(node, ast.alias):
+      name = sc.lookup_name(
+          node.asname if node.asname is not None else node.name)
+      if not name.reads:
+        unused_aliases.add(node)
+
+  unused_imports = []
+  for node in ast.walk(tree):
+    if isinstance(node, ast.Import) or isinstance(node, ast.ImportFrom):
+      for alias in node.names:
+        if alias in unused_aliases:
+          if len(node.names) == 1:
+            unused_imports.append(node)
```

Comment:
This is a bit awkward. I would only return aliases and do the work of removing nodes with no names externally.